### ISSUE TITLE
ARM Architecture release

### ISSUE THREAD

**Issue by gaperez64** [xournalpp/xournalpp#2823]
The current Mac release immediately crashes for me after trying to run it. I suppose it is because of the new architecture (yes, I am running an M1 chip). I compiled from source using the convenient homebrew solution and that indeed works.

I propose to add a release targeting the new architecture for convenience (and for uses who cannot or do not want to compiler Xournal++ themselves).

Additionally, it would be nice to have a build-app script which works on top of the homebrew solution since the current one only works if you use the jhbuild recipe (at least out of the box... it's straightforward to modify it I think).

**Comment by Technius** [comment]
#2390 switches to a macports based build, which should be nicer. I'm not sure if we can do a native ARM build--I don't think Azure Pipelines supports ARM builds for Mac. It's hard for us to support Macs right now because we don't have Mac devices.

**Comment by gaperez64** [comment]
I can try the macports solution once it's merged. For the ARM-based Mac release, I agree that the best would be to wait until Azure pipelines supports the architecture so that it's just a new item in the YAML file.

**Comment by gaperez64** [comment]
I am happy to try and help set up a minimal working version of Xournal++ using homebrew. To this end, here is some more feedback on my side:

- I got the [homebrew version](https://github.com/xournalpp/xournalpp/blob/master/readme/MacBuild.md#xournal-mac-homebrew-build) to compile, though it required to remove "cairo/" from all <cairo/*> includes
- I can run xournalpp after that but I have already found two issues with it:

1. It crashes when I try to export the current document as a pdf. I am attaching both the terminal messages and the crash's log.
[crash.log](https://github.com/xournalpp/xournalpp/files/6027382/crash.log)
[crash-exportpdf.pdf](https://github.com/xournalpp/xournalpp/files/6027384/crash-exportpdf.pdf)
2. When writing using my Wacom tablet (Wacom One) the lines are "squiggly", that is less straight than if I use the builtin trackpad. I am attaching an image of their comparison. There is no special/particular output on the terminal nor do I see error message when writing though so I am at a loss as to what other information I can provide for this. 
<img width="760" alt="Screenshot 2021-02-23 at 08 40 46" src="https://user-images.githubusercontent.com/2659561/108815108-2af20800-75b4-11eb-9a79-60cf3889dad1.png">

**Comment by gaperez64** [comment]
I pulled the most recent source code and recompiled as described above, now the second problem is no longer present :) No idea which recent changes fixed this though. Problem 1 is still present.

**Comment by rolandlo** [comment]
From your bug report in Problem 1, the error happens in line 36 of this piece of code:

https://github.com/xournalpp/xournalpp/blob/e5cdd5f2a15546386e49ed31461f8a394eec1bb0/src/control/jobs/BaseExportJob.cpp#L29-L42

Here is a link to the [reference of `fs::equivalent`](https://en.cppreference.com/w/cpp/filesystem/equivalent)

Try to find what goes wrong here.

**Comment by gaperez64** [comment]
@rolandlo I can reproduce the problem with a smaller C++ file now (see below) which I compiled with `g++ -std=gnu++17 main.cpp -o main`. It seems OS X Big Sur ships with a g++ (= clang++ v13) compiler that has a libc++ not implementing all filesystem operations... I guess the solution is to install a different compiler and library.

```
#include <iostream>
#include <cstdint>
#include <filesystem>
namespace fs = std::filesystem;
int main()
{
    // hard link equivalency
    fs::path p1 = "/Users/guillermoperez/Downloads/aanvraag-ann2.pdf";
    fs::path p2 = "/Users/guillermoperez/Downloads/aanvraag.pdf";
    if (fs::equivalent(p1, p2)) {
        std::cout << p1 << " is equivalent to " << p2 << '\n';
        return 1;
    } else {
        std::cout << p1 << " is NOT equivalent to " << p2 << '\n';
        return 0;
    }   
}
```

**Comment by Technius** [comment]
We have `ghc::filesystem` for this purpose. Perhaps we should force its usage on MacOS?

**Comment by gaperez64** [comment]
I tried two more things:

1. Installed llvm via homebrew and updated the environment variables to have clang++ point to the newly installed binary (and for the libraries). Then, used cmake in my xournal++ `build` directory with `CXX=LOCAL_CLANG cmake ..` followed `make`. I got precisely the same error as reported before. Maybe the binary is still being linked to the libs shipped with OS X?
2. I edited CMakeLists to force the usage of `ghc::filesystem` then recompiled (this time with Apple's clang compiler). This now yields a different error :) I am attaching the [error log](https://github.com/xournalpp/xournalpp/files/6054288/errorlog.20210227-122832.log)
 and the error text.

```
libc++abi: terminating with uncaught exception of type ghc::filesystem::filesystem_error: No such file or directory: '/Users/guillermoperez/Downloads/aanvraag-ann12.pdf', '/Users/guillermoperez/Downloads/aanvraag-ann2.pdf'
```

**Comment by gaperez64** [comment]
With the fix for #2906, I consider all reported problems fixed now! Homebrew builds of xournalpp work almost out of the box for OS X Big Sur on M1, there's just the issue about the Cairo headers but I'll submit a separate issue for that since it's a simple build enhancement.

**Comment by rolandlo** [comment]
@gaperez64 The Cairo headers were fixed in #2919, so homebrew builds should now work out of the box for OS X Big Sur on M1, right?

**Comment by gaperez64** [comment]
@rolandlo Yes, this works out of the box on OS X Big Sur on M1. I just pulled and recompiled to confirm. (Mind you, three libraries are missing from the readme: libsndfile, libzip, and portaudio have to be installed via homebrew too before compiling xournalpp.)

The only reason I have not started working on a homebrew version (bottle is the name I think) is that I still don't have gtk-mac-integration working correctly.
